File: solutions/fortran/binary-search/1/binary_search.f90
module binary_search
  implicit none
contains

  function find(array, val) result(idx)
    integer, dimension(:), intent(in) :: array
    integer, intent(in) :: val
    integer :: idx
    integer :: left, right, mid

    left = 1
    right = size(array)

    do while (left <= right)
      mid = (left + right) / 2
      if (array(mid) == val) then
        idx = mid
        return
      elseif (array(mid) < val) then
        left = mid + 1
      else
        right = mid -1
      end if
    end do

    idx = -1
  end function

end module

File: solutions/csharp/binary-search/1/BinarySearch.cs
using System;

public static class BinarySearch
{
    public static int Find(int[] input, int value)
    {
        var min = 0;
        var max = input.Length -1;

        while (min<= max)
        {
            var mid =(min + max) / 2;
            if (value == input[mid]) {
                return mid;
            } 
            
            if (value < input[mid]) {
                max = mid - 1;
            } else {
                min = mid + 1;
            }
        }

        return -1;
    }
}